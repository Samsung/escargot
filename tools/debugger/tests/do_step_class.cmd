s
bt
s
s
s
s
bt
s
s
s
s
s
s
s
bt
s
s
s
s
s
s
bt
s
s
s
s
s
s
