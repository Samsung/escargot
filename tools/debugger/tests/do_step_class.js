/*
 * Copyright (c) 2021-present Samsung Electronics Co., Ltd
 *
 *  This library is free software; you can redistribute it and/or
 *  modify it under the terms of the GNU Lesser General Public
 *  License as published by the Free Software Foundation; either
 *  version 2.1 of the License, or (at your option) any later version.
 *
 *  This library is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 *  Lesser General Public License for more details.
 *
 *  You should have received a copy of the GNU Lesser General Public
 *  License along with this library; if not, write to the Free Software
 *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
 *  USA
 */

var o = null

class Class1 {
  c1_field1
    = 1
  c1_field2
  [
    "c1_field3"
  ] = 2
  c1_field4 = 3; c1_field5 = 4

  constructor () {
    this.c1_field1 = 5
  }

  static
  c1_sfield1
    = 1
  static c1_sfield2
  static
  [
    "c1_sfield3"
  ] = 2
  static c1_sfield4 = 3; static c1_sfield5 = 4
}

class Class2 extends Class1 {
  #c2_field1
    = 1
  #c2_field2
  #c2_field3 = 2
  #c2_field4 = 3; c2_field5 = 4

  constructor () {
    var v1 = 5
    super()
    this.#c2_field1 = v1
  }

  static
  #c2_sfield1
    = 1
  static #c2_sfield2
  static
  #c2_sfield3
    = 2
  static #c2_sfield4 = 3; static #c1_sfield5 = 4
}

o = new Class2
