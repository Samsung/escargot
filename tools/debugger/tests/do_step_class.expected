Connecting to: localhost:6501
Connection created!!!
Stopped at tools/debugger/tests/do_step_class.js:20
(escargot-debugger) s
Stopped at tools/debugger/tests/do_step_class.js:36 (in function() at line:37, col:7)
(escargot-debugger) bt
tools/debugger/tests/do_step_class.js:37:6 [depth:0]
tools/debugger/tests/do_step_class.js:22:14 [depth:1]
(escargot-debugger) s
Stopped at tools/debugger/tests/do_step_class.js:40 (in function() at line:42, col:7)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_step_class.js:43 (in function() at line:43, col:23)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_step_class.js:43 (in function() at line:43, col:46)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_step_class.js:60 (in function() at line:61, col:7)
(escargot-debugger) bt
tools/debugger/tests/do_step_class.js:61:6 [depth:0]
tools/debugger/tests/do_step_class.js:46:29 [depth:1]
(escargot-debugger) s
Stopped at tools/debugger/tests/do_step_class.js:64 (in function() at line:65, col:7)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_step_class.js:66 (in function() at line:66, col:24)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_step_class.js:66 (in function() at line:66, col:48)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_step_class.js:69
(escargot-debugger) s
Stopped at tools/debugger/tests/do_step_class.js:54 (in constructor() at line:53, col:3)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_step_class.js:55 (in constructor() at line:53, col:3)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_step_class.js:23 (in function() at line:24, col:7)
(escargot-debugger) bt
tools/debugger/tests/do_step_class.js:24:6 [depth:0]
tools/debugger/tests/do_step_class.js:55:5 [depth:2] (in constructor)
tools/debugger/tests/do_step_class.js:69:9 [depth:3]
(escargot-debugger) s
Stopped at tools/debugger/tests/do_step_class.js:26 (in function() at line:28, col:7)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_step_class.js:29 (in function() at line:29, col:15)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_step_class.js:29 (in function() at line:29, col:30)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_step_class.js:32 (in constructor() at line:31, col:3)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_step_class.js:33 (in constructor() at line:31, col:3)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_step_class.js:47 (in function() at line:48, col:7)
(escargot-debugger) bt
tools/debugger/tests/do_step_class.js:48:6 [depth:0]
tools/debugger/tests/do_step_class.js:55:5 [depth:1] (in constructor)
tools/debugger/tests/do_step_class.js:69:9 [depth:2]
(escargot-debugger) s
Stopped at tools/debugger/tests/do_step_class.js:50 (in function() at line:50, col:16)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_step_class.js:51 (in function() at line:51, col:16)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_step_class.js:51 (in function() at line:51, col:31)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_step_class.js:56 (in constructor() at line:53, col:3)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_step_class.js:57 (in constructor() at line:53, col:3)
(escargot-debugger) s
Connection closed.
